src/Requests/MutationRequest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Intermax\LaravelJsonApi\Requests;
+
+use Exception;
+use Illuminate\Contracts\Container\Container;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Support\Collection;
+
+abstract class MutationRequest extends FormRequest
+{
+    abstract protected function type(): string;
+
+    public function authorize(): bool
+    {
+        if (! $this->hasHeader('Content-Type') || ! str_starts_with((string) $this->header('Content-Type'), 'application/vnd.api+json')) {
+            abort(415, 'Invalid content type');
+        }
+
+        return true;
+    }
+
+    /**
+     * @return array<string, mixed>
+     *
+     * @throws Exception
+     */
+    public function rules(Container $app): array
+    {
+        $rules = [
+            'data.id' => ['string'],
+            'data.type' => ['required', 'in:'.$this->type(), 'string'],
+        ];
+
+        // 'id' is required in JSON:API specification when updating a resource.
+        if ($this->method() !== 'POST') {
+            $rules['data.id'] = array_merge(['required'], $rules['data.id']);
```

Comment:
I think this would be easier to read:
```suggestion
            $rules['data.id'][] = 'required';
```